documentation/rest-clients.md

```diff
@@ -0,0 +1,158 @@
+import { DefaultAzureCredential } from "@azure/identity";
+
+const client = ExampleClient("https://example.org/", new DefaultAzureCredential());
+
+// Send a GET request to https://example.org/hello
+const response = await client.path("/hello").get();
+
+if(response.status !== "200") {
+  throw response.body.error;
+}
+
+console.log(response.body);
+// {content: "Hello World"}
+```
+
+## What is a REST Client
+Our REST clients provide a simple, reliable connection to raw HTTP. We provide a `path` function on the client to set the resource to target and methods for each verb it supports. Using `path` provides strong types to check for the validity of the path, parameters, and response types.
+
+We also expose a `pathUnchecked` function which allows targeting an arbitrary path. We can provide some help for detecting required path parameters, but responses and body parameters are loosely typed.
+
+Both methods fully harness the power of `@azure/core-rest-pipeline` and `@azure/identity`.
+
+The basic structure of calls with REST clients is:
+
+1. Initialize your client
+2. Send the request
+3. Handle the response
+
+We will go into each step in the following sections
+
+## 1. Initialize the Client
+First import the client
+
+```typescript
+import ExampleClient from "@azure-rest/example-client";
+```
+
+Most clients require authenticating through their `credential` parameter. Depending on what authentication support your library is using, you can either authenticate with AAD or authenticate with an AzureKeyCredential.
+
+Additionally, most of our clients accept an `endpoint` parameter at initialization, usually a link to your own resource.
+
+### Authenticating with AAD
+Authenticating with AAD
+Depending on your library, our clients support authenticating with an Azure Active Directory (AAD) token credential. We always recommend using a credential type obtained from the `@azure/identity` library for AAD authentication. For this example, we use the most common DefaultAzureCredential.
+
+As an installation note, the `@azure/identity` library is not a dependency of this library. Please run `npm install @azure/identity` before using AAD authentication
+
+The following code snippet shows you how to authenticate with a DefaultAzureCredential.
+
+```typescript
+import ExampleClient from "@azure-rest/example-client";
+import { DefaultAzureCredential } from "@azure/identity";
+
+const client = ExampleClient("https://example.org/", new DefaultAzureCredential());
+```
+
+## 2. Send a request
+
+Once the client has been initialized we need to set a path to work with. For this, the REST client exposes 2 functions `path` and `pathUnchecked`
+
+### Path
+Path function takes a path as the first parameter and accepts any path defined in the Swagger. It also detects if the path needs a path parameter and requires them as positional parameters to `path`. Once the path is set users can access functions for all the supported verbs on that path
```

Comment:
I would try not to use "swagger" here if possible, as hopefully most folks will be blissfully unware that swagger exists.  Perhaps we could say something about how the SDK has a predefined list of routes to resources provide by the service, but not dive into the details of how it got that?

Reply:
+1 with Matt, Swagger is an implementation details. For instance:
```Path function takes a path as the first parameter and accepts any path the service would accept```

Reply:
Yes! This is a very good point. What about this?

> The `path` function takes a string as the first parameter and accepts any path documented by the service, this function will help with autocomplete to discover all available paths. It also detects if the path needs parameters and makes them required positional parameters to `path`. Once the path is set users can access functions for all the supported verbs on that path

Reply:
Love it!